### PR TITLE
Feature: refactor quick panel for (mostly) static options

### PR DESCRIPTION
**Pull request by asfaltboy**
This  PR adds two reusable mixins that replace much repetitive and sometimes inconsistent code. 
Hopefully this way the commands has less cruft and easier to read.

In  this refactor I only touched panels with "static options". The next step would be to replace the remaining panels that prompt for branch/remote. By default most of the updated commands run in a synchronous manner (in the current thread), since the panel options are available immediately.

### CODE REVIEW

**Comment by stoivo** [comment]
This looks really great.

**Comment by randy3k** [comment]
Look great. Thanks.

**Comment by randy3k** [comment]
It seems that `git: log current file` and `git: graph current file` are broken.

**Comment by asfaltboy** [comment]
@randy3k thanks! Silly mistake with loop local variables. I fixuped the specific GsLogCommand commit, hence had to force push all the last 5 ones.

File: core/commands/log_graph.py [hunk 1]
```diff
@@ -152,81 +152,59 @@ def get_graph_args(self):
...
+            self.actions.extend([
                 ["diff_commit", "Diff commit"],
-                ["diff_commit_cache", "Diff commit (cached)"]
-            ]
+                ["diff_commit_cache", "Diff commit (cached)"],
+            ])
 
-        self._file_path = self.file_path
         if self._file_path:
             self.actions.insert(1, ["show_file_at_commit", "Show file at commit"])
```
Comment (randy3k) [inline core/commands/log_graph.py]: For some reason, it is still not functioning. This option does no show up in `git: graph current file`

Comment (randy3k) [inline core/commands/log_graph.py]: I figured out the reason. We need to evaluate `self.file_path` before the `graph_view` is created.

Comment (asfaltboy) [inline core/commands/log_graph.py]: Sort of, I just forgot to pass the evaluated `self.file_path` property to the parent class, so it was using default `None` .. should be fixed now

**Comment by randy3k** [comment]
Great, it fixes it.

**Pull request merged**